### PR TITLE
API can provide list of products

### PR DESCRIPTION
**Pull request by ViaMarcus**
### Show menu items
This PR adds the ability for the API to serve menu items from the database.
It also adds the menu_item model, and tests for the model.
The endpoint is GET '/api/v1/menu_items'
It takes no params and returns
`{ items: [<MenuItem>,<MenuItem>]}` where each `MenuItem` has id, name, description and price.
[PT Card](https://www.pivotaltracker.com/story/show/172642043)

### CODE REVIEW

**Review by kermit-klein — COMMENTED** [review]
It works!

File: app/controllers/api/v1/menu_items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Api::V1::MenuItemsController < ApplicationController
+	def index
```
Comment (oliverochman) [inline app/controllers/api/v1/menu_items_controller.rb]: please fix indentation in this file

File: spec/requests/api/v1/menu_items_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::MenuItemsController, type: :request do
+    let!(:food1) { create(:menu_item, name: 'Pickled Egg', description: 'An egg that is in a reeeeal pickle!', price: 69) }
```
Comment (oliverochman) [inline spec/requests/api/v1/menu_items_spec.rb]: you dont test any of the attributes for these menu items. 

please refactor it to

```
let!(:menu_items) { 3.times { create(:menu_item) } }
```

File: spec/requests/api/v1/menu_items_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    describe 'GET /v1/menu_items' do
+        before do
+            get '/api/v1/menu_items'
+        end
+
+        it 'should return a 200 response' do
+            expect(response).to have_http_status 200
+        end
+
+        describe 'should return a response with' do
```
Comment (oliverochman) [inline spec/requests/api/v1/menu_items_spec.rb]: no reason to have these tests in a separate describe block.

File: spec/requests/api/v1/menu_items_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            let(:json_response) { JSON.parse(response.body) }
```
Comment (oliverochman) [inline spec/requests/api/v1/menu_items_spec.rb]: please extract this helper method from this file

File: spec/requests/api/v1/menu_items_spec.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+            it 'should return a list of menu items' do
+                expect(json_response["items"].length).to eq 3
+            end
+
+            it 'should return items with a name' do
+                expect(json_response["items"][0]["name"]).to eq food1.name
```
Comment (oliverochman) [inline spec/requests/api/v1/menu_items_spec.rb]: instead of testing what it equals to you can test that it includes `["name"]`

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/menu_items_controller.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+		items = MenuItem.all
+		render json: { items: items }
```
Comment (oliverochman) [inline app/controllers/api/v1/menu_items_controller.rb]: if the models name is menu items, you should name the key to that as well

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by hashtagDora — COMMENTED** [review]
I will do the changes

**Review by oliverochman — APPROVED** [review]

**Pull request merged**